utilities_common/multi_asic.py

```diff
@@ -0,0 +1,138 @@
+import argparse
+import functools
+
+import click
+from sonic_py_common import multi_asic
+from utilities_common import constants
+
+
+class MultiAsic(object):
+
+    def __init__(self, display_option=constants.DISPLAY_ALL,
+                 namespace_option=None):
+        self.namespace_option = namespace_option
+        self.display_option = display_option
+        self.current_namespace = None
+        self.is_multi_asic = multi_asic.is_multi_asic()
+
+    def is_object_internal(self, object_type, cli_object):
+        '''
+        The function checks if a CLI object is internal and returns true or false.
+        Internal objects are port or portchannel which are connected to other 
+        ports or portchannels within a multi ASIC device.
+
+        For single asic, this function is not applicable
+        '''
+        if object_type == constants.PORT_OBJ:
+            return multi_asic.is_port_internal(cli_object)
+        elif object_type == constants.PORT_CHANNEL_OBJ:
+            return multi_asic.is_port_channel_internal(cli_object)
+        elif object_type == constants.BGP_NEIGH_OBJ:
+            return multi_asic.is_bgp_session_internal(cli_object)
+
+    def skip_display(self, object_type, cli_object):
+        '''
+        The function determines if the passed cli_object has to be displayed or not.
+        returns true if the display_option is external and  the cli object is internal.
+        returns false, if the cli option is all or if it the platform is single ASIC.
+
+        '''
+        if not self.is_multi_asic:
+            return False
+        if self.display_option == constants.DISPLAY_ALL:
+            return False
+        return self.is_object_internal(object_type, cli_object)
+
+    def get_ns_list_based_on_options(self):
+        ns_list = []
+        if not self.is_multi_asic:
+            return [constants.DEFAULT_NAMESPACE]
+        else:
+            namespaces = multi_asic.get_all_namespaces()
+            if self.namespace_option is None:
+                if self.display_option == constants.DISPLAY_ALL:
+                    ns_list = namespaces['front_ns'] + namespaces['back_ns']
+                else:
+                    ns_list = namespaces['front_ns']
+            else:
+                if self.namespace_option not in namespaces['front_ns'] and \
+                        self.namespace_option not in namespaces['back_ns']:
+                    raise ValueError(
+                        'Unknown Namespace {}'.format(self.namespace_option))
+                ns_list = [self.namespace_option]
+        return ns_list
+
+
+def multi_asic_ns_choices():
```

Comment:
This API may not be required as multi_asic.get_namespace_list() already gives the namespacelist correctly. But, it is ok for now - could remove later as well.